pkg/server/plugin/upstreamauthority/gcpcas/gcpcas.go

```diff
@@ -141,6 +142,9 @@ func (p *Plugin) Configure(ctx context.Context, req *configv1.ConfigureRequest)
 	if config.RootSpec.Location == "" {
 		return nil, status.Error(codes.InvalidArgument, "configuration has empty root_cert_spec.Location property")
 	}
+	if config.RootSpec.CaPool == "" {
```

Comment:
It looks like we could have the plugin create a CA pool, but I think that may be out of scope.  If the v1beta1 API is gone we really need to make the change.  Are other users on GCP having the same issue?

Reply:
I don't like the idea of having the plugin create the pool.  However there's a few different ways of varying efficiency I can think of to accomplish this:

1.  Search every ca pool in the project & region for a certificate authority with the matching labels
2.  Enforce that the ca pool must have the same labels and search for the ca pool with the defined labels and a root in the pool with the same labels.

Reply:
@amartinezfayo Any preference or other ideas from the maintainers on the preferred way to handle the backward compatibility issue?

Personally I think making ca_pool optional and otherwise searching through every CA pool in the provided region for a matching CA makes sense and acts in accordance with the beta functionality.

Reply:
I'm not sure how onerous that search is but that seems like a fine approach to me provided that we:
1. Fail if more than one CA is matched
2. Issue a warning when ca_pool is not configured

The warning could be something like:
```
// TODO: remove deprecated fallback behavior in 1.2
log.Warn("The ca_pool value is not configured. Falling back to searching the region for matching CAs. The ca_pool configurable will be required in a future release.")
```

Reply:
Actually, we can't fail if more than one is matched. That breaks the current behavior of adding the rest to the trust bundle.

Reply:
Or maybe we just fail if the set of matched CAs spans more than one pool?

Reply:
This seems to align pretty closely with what my understanding of a CA pool is. Also seems that at some later point we want to instead sign using a pool and not require labels, but I'm new to this GCP CAS thing.

Reply:
Failing if the set of matched CAs spans more than one pool makes sense to me.